src/main/java/net/minecraftforge/fml/common/network/handshake/NetworkDispatcher.java

```diff
@@ -199,12 +200,14 @@ public void update()
         NBTTagCompound playerNBT = scm.getPlayerNBT(player);
         if (playerNBT!=null)
         {
-            return playerNBT.getInteger("Dimension");
-        }
-        else
-        {
-            return 0;
+            int dimension = playerNBT.getInteger("Dimension");
+            World playerWorld = scm.getServerInstance().worldServerForDimension(dimension);
+            if (playerWorld != null)
```

Comment:
Dimension manager would be the place to ask, this force loads the dimension which is bad.